### PR TITLE
Customizable send_mail

### PR DESCRIPTION
**Pull request by abulte**
Hi,

This pretty straightforward PR lets you override the `send_mail` method with your own logic. The override method is similar to `render_template`.

Thanks for considering this!

### CODE REVIEW

File: docs/customizing.rst [hunk 1]
```diff
@@ -213,6 +213,36 @@ Celery. The practical way with custom serialization may look like so::
...
+        def custom_send_mail(subject, recipient, template, **context):
+            # implement your own logic here
+            pass
+
+        datastore = SQLAlchemyUserDatastore(db, User, Role)
+        security_ctx.send_mail = custom_send_mail
+
+        return app
+
+Note that the above ``security.send_mail_task`` override will be useless if you
```
Comment (jirikuncar) [inline docs/customizing.rst]: Have you consider using `.. note::` directive http://www.sphinx-doc.org/en/stable/markup/para.html#directive-note ?

File: docs/customizing.rst [hunk 2]
```diff
@@ -213,6 +213,36 @@ Celery. The practical way with custom serialization may look like so::
...
+    security = Security()
+
+    def create_app(config):
+        """Initialize Flask instance."""
+
+        app = Flask(__name__)
+        app.config.from_object(config)
+
+        def custom_send_mail(subject, recipient, template, **context):
+            # implement your own logic here
```
Comment (jirikuncar) [inline docs/customizing.rst]: can you add an example when it would be necessary to override the `send_mail` method?

**Comment by abulte** [comment]
@jirikuncar I added an example. I hope it does not feel too contrived. My real world use case is too complex to be suitable for an example.

File: docs/customizing.rst [hunk 3]
```diff
@@ -213,6 +213,44 @@ Celery. The practical way with custom serialization may look like so::
...
+-----------------------
+
+It's also possible to completely override the ``security.send_mail`` method to
+implement your own logic.
+
+For example, you might want to use an alternative email library like `Flask-Emails`:
+
+    from flask import Flask
+    from flask_security import Security, SQLAlchemyUserDatastore
+    from flask.ext.emails import Message
```
Comment (jirikuncar) [inline docs/customizing.rst]: `from flask_emails ...`

**Review by jirikuncar — APPROVED** [review]

**Comment by fuhrysteve** [comment]
Thought I'd mention that I personally have been using this branch of `Flask-Mail` to override recipient in test environments via configuration: `MAIL_CATCH_ALL = 'dev@mycompany.com'`

https://github.com/mattupstate/flask-mail/pull/118

It seems to me that `Flask-Mail` should either be maintained, or dropped by `Flask-Security`

**Pull request merged**